### PR TITLE
build: clean up the gn template for rust targets a bit

### PR DESCRIPTION
**Pull request by piscisaureus**

### CODE REVIEW

File: build_extra/rust/rust.gni [hunk 1]
```diff
@@ -47,74 +47,124 @@ rust_test_ldflags = exec_script("get_rust_ldflags.py",
...
+          rlib = "$out_dir/lib$crate_name.rlib"
+        },
+      ]
+    }
+  }
+  if (defined(invoker.extern_version)) {
+    foreach(info, invoker.extern_version) {
+      extern_rlibs += [
+        {
+          crate_suffix = exec_script("//tools/sha256sum.py",
```
Comment (ry) [inline build_extra/rust/rust.gni]: Aside: I suspect, without evidence, that this exec_script is slowing incremental builds significantly.

Comment (piscisaureus) [reply build_extra/rust/rust.gni]: Why? It runs in the `gn gen` phase, which takes 2 seconds on my laptop.
Or am I missing something?

Comment (piscisaureus) [reply build_extra/rust/rust.gni]: Also I'd be more than happy to get rid of it if there was a way...

**Comment by piscisaureus** [comment]
This is very low priority but it was just sitting there. Did it when I was trying out Alex' suggestions.

**Pull request merged**